From 36927554d9768dfdce12c6ba87b9e63270b9f137 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Wed, 27 Nov 2013 13:54:40 +0800
Subject: [PATCH 263/602] ENGR00289827 Enable charge mode for sabresd board

Start charger service based on charge mode.
Make system into low power mode in charge mode.
Append "androidboot.mode=charger" into bootargs will make
system into charge mode.

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 imx6/imx6.mk        |  4 ++++
 sabresd_6dq/init.rc | 15 +++++++++++++++
 2 files changed, 19 insertions(+)

diff --git a/imx6/imx6.mk b/imx6/imx6.mk
index baf273f..1ada99e 100644
--- a/imx6/imx6.mk
+++ b/imx6/imx6.mk
@@ -308,6 +308,10 @@ PRODUCT_PACKAGES += \
 	libreference-ril-zte.so \
 	libruntime-ril-port
 
+PRODUCT_PACKAGES += \
+    charger_res_images \
+    charger
+
 PRODUCT_COPY_FILES +=	\
 	device/fsl/common/input/Dell_Dell_USB_Keyboard.kl:system/usr/keylayout/Dell_Dell_USB_Keyboard.kl \
 	device/fsl/common/input/Dell_Dell_USB_Keyboard.idc:system/usr/idc/Dell_Dell_USB_Keyboard.idc \
diff --git a/sabresd_6dq/init.rc b/sabresd_6dq/init.rc
index eb9661c..e8f962c 100755
--- a/sabresd_6dq/init.rc
+++ b/sabresd_6dq/init.rc
@@ -108,6 +108,18 @@ on boot
 # mount the debugfs
     mount debugfs none /sys/kernel/debug/
 
+on charger
+    # Enable Power modes and set the CPU Freq Sampling rates
+    write /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor "powersave"
+    write /sys/devices/system/cpu/cpufreq/ondemand/up_threshold 90
+    write /sys/devices/system/cpu/cpufreq/ondemand/sampling_rate 50000
+    write /sys/devices/system/cpu/cpufreq/ondemand/io_is_busy 1
+    write /sys/devices/system/cpu/cpufreq/ondemand/sampling_down_factor 4
+    write /sys/devices/system/cpu/cpu1/online 0
+    write /sys/devices/system/cpu/cpu2/online 0
+    write /sys/devices/system/cpu/cpu3/online 0
+
+
 service p2p_supplicant /system/bin/wpa_supplicant \
     -ip2p0 -Dnl80211 -c/data/misc/wifi/p2p_supplicant.conf -N \
     -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf
@@ -147,6 +159,9 @@ service ing /system/bin/ingsvcd -c /system/etc/gps/Orion.ini
     group gps
     oneshot
 
+service charger /charger
+    class charger
+
 on fs
 # mount ext4 partitions
     mount_all /fstab.freescale
-- 
1.8.0

